relay-metrics/src/aggregation.rs

```diff
@@ -764,6 +776,43 @@ impl AggregatorConfig {
         Duration::from_secs(self.debounce_delay)
     }
 
+    /// Determines the target bucket for an incoming bucket timestamp and bucket width.
+    ///
+    /// We select the output bucket which overlaps with the center of the incoming bucket.
+    /// Fails if timestamp is too old or too far into the future.
+    fn get_bucket_timestamp(
+        &self,
+        timestamp: UnixTimestamp,
+        bucket_width: u64,
+    ) -> Result<UnixTimestamp, AggregateMetricsError> {
+        // We know this must be UNIX timestamp because we need reliable match even with system
+        // clock skew over time.
+
+        let now = UnixTimestamp::now().as_secs();
+        let min_timestamp = UnixTimestamp::from_secs(now - self.max_age);
+        let max_timestamp = UnixTimestamp::from_secs(now + self.max_future);
```

Comment:
These arithmetic operations could overflow, but only if someone configures bogus values for `max_age`/`max_future`, so I guess a panic is fine in this case.